script.js

```diff
@@ -1,4 +1,182 @@
+    }
+    if (checkNumbers(splitNumbers) === false) {
+      return `ERROR 3.Please input ${totalDice} numbers ONLY! And they should not have any repeats! The numbers are ${recentRolls}`;
+    }
+    if (playerTurn === 1) {
+      for (let i = 0; i < splitNumbers.length; i += 1) {
+        player1Hand += recentRolls[splitNumbers[i] - 1];
+      }
+    } else {
+      for (let i = 0; i < splitNumbers.length; i += 1) {
+        player2Hand += recentRolls[splitNumbers[i] - 1];
+      }
+    }
+    let message = "";
+    if (playerTurn === 1) {
+      message = `${callPlayer(
+        playerTurn
+      )} has opted for ${player1Hand}! Press Submit to proceed to Player 2's turn!`;
+    } else {
+      message = `${callPlayer(
+        playerTurn
+      )} has opted for ${player2Hand}! Press Submit to head towards the resolution!`;
+    }
+    if (playerTurn === 1) {
+      playerTurn = 2;
+    } else {
+      playerTurn = 1;
+    }
+    gameState += 1;
+    recentRolls = [];
+    return message;
+  } else {
+    gameState = 0;
+    winnerMessage = compareScores(player1Hand, player2Hand);
+    player1Hand = "";
+    player2Hand = "";
+    return winnerMessage;
+  }
+}
+
+// // differentiating between the player
+
+function callPlayer(playerTurn) {
+  if (playerTurn === 1) {
+    return "Player 1";
+  } else {
+    return "Player 2";
+  }
+}
+
+//roll2dice
+
+function rollDice() {
+  for (let i = 0; totalDice > i; i += 1) {
+    recentRolls.push(Math.ceil(Math.random() * 6));
+  }
+  console.log(recentRolls);
+  return;
+}
+// a function to show options
+function showOptions(playerRolls) {
+  return `Hello ${callPlayer(
+    playerTurn
+  )}! Your rolls are ${playerRolls}. Please key in the order you prefer the dice to go in! (Example: 2431 means 2nd dice, 4th dice, 3rd dice then 1st dice respectively!)`;
+}
+
+// checking duplicates
+function checkDupes(testsubject) {
+  for (let i = 0; i < testsubject.length; i++) {
+    for (let j = i + 1; j < testsubject.length; j++) {
+      if (testsubject[i] === testsubject[j]) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+//comparing scores and setting it to
+function compareScores(player1Hand, player2Hand) {
+  let winner = "";
+  Number(player1Hand);
+  Number(player2Hand);
+  if (player1Hand === player2Hand) {
+    return `It is a draw! Both players have rolled ${player1Hand}. Hit submit to reset the game!`;
+  } else {
+    if (player1Hand > player2Hand) {
+      winner = "Player 1";
+    } else {
+      winner = "Player 2";
+    }
+    return `${winner} has won! Player 1 rolled ${player1Hand} while Player 2 rolled ${player2Hand}. Hit submit to reset the game!`;
+  }
+}
+
+//function validP1P3
+
+function validP1P3(input) {
+  return (
+    !isNaN(input) &&
+    input !== "" &&
+    input.length === Number(totalDice) &&
+    input.length !== 0 &&
+    input !== undefined
+  );
+}
+
+function checkNumbers(input) {
```

Comment:
This function could potentially return undefined. Always make sure your function returns some actual value